src/components/dragBoxLayer.ts

```diff
@@ -416,6 +428,32 @@ export module Components {
       this._setMovableClass();
       return this;
     }
+
+    public destroy() {
+      super.destroy();
+      this._dragStartCallbacks.forEach((callback) => this._dragCallbacks.delete(callback));
+      this._dragCallbacks.forEach((callback) => this._dragCallbacks.delete(callback));
+      this._dragEndCallbacks.forEach((callback) => this._dragEndCallbacks.delete(callback));
+      this._disconnectInteraction();
+    }
+
+    public detach() {
+      this._resetBound();
+      this._dragInteraction.detachFrom(this);
+      return super.detach();
+    }
+
+    public anchor(selection: d3.Selection<void>) {
+      this._dragInteraction.attachTo(this);
+      return super.anchor(selection);
+    }
+
+    private _resetBound() {
```

Comment:
Can we call it `_resetState()`, as ideally, we can make more resets here. 

By doing so, we make it more obvious that reseting the bounds is done for this purpose (resetting the state). Otherwise setting both bounds to 0 and the method `_resetBound()` are synonyms, so we don't get any benefit.